app/models/friendship.rb

```diff
@@ -1,4 +1,18 @@
+# frozen_string_literal: true
+
 class Friendship < ApplicationRecord
-    belongs_to :friend_active, class_name "user"
-    belongs_to :friend_passive, class_name "user"
+  # associations
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  # validations
+  validates :status, inclusion: { in: [true, false] }
+
+  # scopes
+  scope :already, ->(user, friend) { where('user_id = (?)', user).where('friend_id = (?)', friend) }
+  scope :invitation_requests, ->(user) { where('friend_id = (?)', user).where('status = (?)', false) }
```

Comment:
- [ ] There is some sort of logic duplication in these scopes. For example:
- `invitation_requests` scope can be achieved with `user.passive_friendships`
- `invitation_sent` scsope can also be achieved with `user.active_friendships`

I suggest to use association because defining associations can give your number of other helper methods like `passive_friendship_ids` which will return an array of friendships records ids associated with a particular user.

Also `passive_friendships` return value acts like an array provided also with the helper methods (though it is not actually an array), so you can do something like `user.passive_friendships << [friendship_object]`.

Reply:
how could I use associations for that purpose?

Reply:
@samgaco Sorry, what do you mean?